### PR TITLE
Update Dashboard to use V2 SDB API

### PR DESCRIPTION
**Pull request by sdford**
* Updates the Dashboard to take IAM principal ARNs for SDB permissions, instead of account id and role name

### CODE REVIEW

File: app/components/CreateSDBoxForm/validator.js [hunk 1]
```diff
@@ -57,14 +57,11 @@ const validateUserGroupPermissions = (permission, index, errors) => {
...
-        errors.iamRolePermissions[`${index}`].accountId = 'Required'
-    } else if (! /^\d+$/.test(permission.accountId)) {
-        errors.iamRolePermissions[`${index}`].accountId = 'Numeric only'
-    }
 
-    if (! permission.iamRoleName) {
-        errors.iamRolePermissions[`${index}`].iamRoleName = 'You must enter a IAM role name for this permission'
+    if (! permission.iamPrincipalArn) {
+        errors.iamRolePermissions[`${index}`].iamPrincipalArn = 'You must enter a IAM principal ARN for this permission'
+    } else if (! /^arn:aws:(iam|sts)::.+$/.test(permission.iamPrincipalArn)) {
```
Comment (tlisonbee) [inline app/components/CreateSDBoxForm/validator.js]: Pull request looks good to me.   My only question is how did you get this regex?  Do we know this is right?  Can we test it somehow?

It seems like it is good to have some regex here but just curious if it is flexible enough to allow everything that should be allowed, etc.

Comment (sdford) [inline app/components/CreateSDBoxForm/validator.js]: @tlisonbee So I generated this regex based off of this documentation:
http://docs.aws.amazon.com/IAM/latest/UserGuide/reference_identifiers.html

I tested a few of the examples in that doc manually in the dashboard. I also have the same regex in CMS where I wrote unit tests to make sure that those examples pass as valid ARNs.

**Comment by fieldju** [comment]
@sdford weren't you going to refactor the component names to reflect the fact that they are for all IAM principals types rather than just IAM roles.

**Comment by sdford** [comment]
@fieldju I can do the component names. Those should be easy, but the variable names will be a more involved because CMS is still passing an 'iam_role_permissions' field in a SDB object.

**Comment by fieldju** [comment]
@sdford we can talk about it later, this is fine for now.

**Pull request merged**